server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -712,7 +724,7 @@ private void finalizeSnapshotDeletionFromPreviousMaster(ClusterChangedEvent even
      * snapshot and remove it from the cluster state.
      */
     private void removeFinishedSnapshotFromClusterState(ClusterChangedEvent event) {
-        if (event.localNodeMaster() && !event.previousState().nodes().isLocalNodeElectedMaster()) {
+        if (event.previousState().nodes().isLocalNodeElectedMaster() == false) {
```

Comment:
What is this change?

Reply:
Wops, that snuck in from another refactoring, sorry about that. The check for `localNodeMaster()` here is redundant because it already happens upstream. I'll revert this here before merging though and put it in another PR.

Reply:
reverted in https://github.com/elastic/elasticsearch/pull/37629/commits/0880cebd69d5ae83cd6eee734d5fd9a77dc476b4